src/browser_action/render_scripts.js

```diff
@@ -127,23 +131,28 @@ const renderScripts = async function () {
   scriptsDiv.textContent = '';
 
   const installedScripts = await getInstalledScripts();
-  const { enabledScripts = [] } = await browser.storage.local.get('enabledScripts');
+  const { enabledScripts = [], specialAccess = [] } = await browser.storage.local.get();
 
   const orderedEnabledScripts = installedScripts.filter(scriptName => enabledScripts.includes(scriptName));
   const disabledScripts = installedScripts.filter(scriptName => enabledScripts.includes(scriptName) === false);
 
   for (const scriptName of [...orderedEnabledScripts, ...disabledScripts]) {
     const url = getURL(`/scripts/${scriptName}.json`);
     const file = await fetch(url);
-    const { title = scriptName, description = '', icon = {}, help = '', relatedTerms = [], preferences = {} } = await file.json();
+    const { title = scriptName, description = '', icon = {}, help = '', relatedTerms = [], preferences = {}, deprecated = false } = await file.json();
 
     const scriptTemplateClone = document.getElementById('script').content.cloneNode(true);
 
     const detailsElement = scriptTemplateClone.querySelector('details.script');
     detailsElement.dataset.relatedTerms = relatedTerms;
+    detailsElement.dataset.deprecated = deprecated;
 
     if (enabledScripts.includes(scriptName) === false) {
       detailsElement.classList.add('disabled');
+
+      if (deprecated && !specialAccess.includes(scriptName)) {
+        detailsElement.hidden = true;
```

Comment:
Any reason to keep the element in the DOM?

Personally I would just skip rendering it:

```suggestion
        continue;
```

Reply:
Not particularly! I regularly remove the `display: none` css on deprecated XKit 7 scripts to install them for testing purposes, but in this repo it's quite easy to just add a script name to `enabledScripts` and I assume a user dedicated/knowledgeable enough to do one will just as easily do the other.